### DISCUSSION TITLE
可搜索的选择器（nz-select） 是如何让下拉 点击别的地方隐藏起来的？

### DISCUSSION THREAD

**Discussion by John0King** [NG-ZORRO/ng-zorro-antd#6971] [Q&A]
我正在用另外一个框架 baidu的 san  ,   想要做一个 flyout 组件，  主要目的就是做成 nz-select 这样的 （不要问为啥不用 angular, 问了都是泪， 什么年代了，居然还用IE8） 
但是有个难题一直困扰着我：
focus 打开，或者click打开后 ， 怎么让他点击其他地方或者 blur 的时候 把 flyout 关闭 ， 但是点击 flyout上的元素还得不能关闭，
点击 flyout上的 [item] 后，才能关闭

**Comment by wzhudev** [comment]
zorro 的这项功能基于 Angular CDK 的能力，例如：

https://github.com/NG-ZORRO/ng-zorro-antd/blob/78c16a2794f5e5c2f5098c83ea540c88dd9d6d98/components/tooltip/tooltip.ts#L95

实际上就是检测 click 事件的 target 元素是否被 layout 所包含，如果不包含就关闭面板，如果点击到了 item 也关闭面板，其他情况不关闭。
